src/bootstrap/triple.rs

```diff
@@ -0,0 +1,126 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::convert;
+use std::ffi;
+use std::fmt;
+use std::path;
+
+/// Platform represented by a hyphen-delimited string. Does not
+/// necessarily consist of three items. e.g.:
+///
+/// * `i686-apple-darwin`
+/// * `powerpc-unknown-linux-gnu`
+#[derive(Clone, Debug, Default, Eq, Hash, PartialEq, RustcDecodable)]
+pub struct Triple(pub String);
```

Comment:
In the future, maybe this could be an `enum` that has all the platforms we support?